src/vs/workbench/contrib/debug/browser/replFilterView.ts

```diff
@@ -0,0 +1,180 @@
+		@IContextViewService private readonly contextViewService: IContextViewService) {
+		super(null, action);
+		this.delayedFilterUpdate = new Delayer<void>(200);
+		this._register(toDisposable(() => this.delayedFilterUpdate.cancel()));
+	}
+
+	render(container: HTMLElement): void {
+		this.container = container;
+		DOM.addClass(this.container, 'repl-panel-action-filter-container');
+
+		this.element = DOM.append(this.container, DOM.$(''));
+		this.element.className = this.class;
+		this.createInput(this.element);
+		this.updateClass();
+	}
+
+	focus(): void {
+		if (this.filterInputBox) {
+			this.filterInputBox.focus();
+		}
+	}
+
+	private clearFilterText(): void {
+		if (this.filterInputBox) {
+			this.filterInputBox.value = '';
+		}
+	}
+
+	private createInput(container: HTMLElement): void {
+		this.filterInputBox = this._register(this.instantiationService.createInstance(ContextScopedHistoryInputBox, container, this.contextViewService, {
+			placeholder: localize('workbench.debug.filter.placeholder', "Filter. E.g.: text, !exclude"),
+			history: this.filters.filterHistory
+		}));
+		this.filterInputBox.value = this.filters.filterText;
+		this._register(this.filterInputBox.onDidChange(() => this.delayedFilterUpdate.trigger(() => this.onDidInputChange(this.filterInputBox!))));
+		this._register(this.filters.onDidChange((event: IReplFiltersChangeEvent) => {
+			if (event.filterText) {
+				this.filterInputBox!.value = this.filters.filterText;
+			}
+		}));
+		this._register(DOM.addStandardDisposableListener(this.filterInputBox.inputElement, DOM.EventType.KEY_DOWN, (e: any) => this.onInputKeyDown(e)));
+		this._register(DOM.addStandardDisposableListener(container, DOM.EventType.KEY_DOWN, this.handleKeyboardEvent));
+		this._register(DOM.addStandardDisposableListener(container, DOM.EventType.KEY_UP, this.handleKeyboardEvent));
+		this._register(DOM.addStandardDisposableListener(this.filterInputBox.inputElement, DOM.EventType.CLICK, (e) => {
+			e.stopPropagation();
+			e.preventDefault();
+		}));
+		this._register(this.filters.onDidChange(e => this.onDidFiltersChange(e)));
+	}
+
+	private onDidFiltersChange(e: IReplFiltersChangeEvent): void {
+		if (e.layout) {
+			this.updateClass();
+		}
+	}
+
+	private onDidInputChange(inputbox: HistoryInputBox) {
+		inputbox.addToHistory();
+		this.filters.filterText = inputbox.value;
+		this.filters.filterHistory = inputbox.getHistory();
+	}
+
+	// Action toolbar is swallowing some keys for action items which should not be for an input box
+	private handleKeyboardEvent(event: StandardKeyboardEvent) {
+		if (event.equals(KeyCode.Space)
+			|| event.equals(KeyCode.LeftArrow)
+			|| event.equals(KeyCode.RightArrow)
+			|| event.equals(KeyCode.Escape)
+		) {
+			event.stopPropagation();
+		}
+	}
+
+	private onInputKeyDown(event: StandardKeyboardEvent) {
+		let handled = false;
+		if (event.equals(KeyCode.Escape)) {
+			this.clearFilterText();
+			handled = true;
+		}
+		if (handled) {
```

Comment:
I believe this should just be in the previous `if` statment and we do not need the `handled` boolean. Feels like a copy paste error.